addons/agent_mirrorpeer_controller.go

```diff
@@ -181,6 +183,29 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 
 		logger.Info("Labeled the default VolumeSnapshotClasses successfully")
+
+		if mirrorPeer.Annotations == nil {
+			mirrorPeer.Annotations = make(map[string]string)
+		}
+
+		spokeKeyTypeAnn := fmt.Sprintf("%s.%s", r.SpokeClusterName, SpokeKeyTypeAnnotation)
+		if mirrorPeer.Annotations[KeyTypeAnnotation] != KeyTypeAES256k {
+			spokeAnnVal, err := r.updateSpokeKeyType(ctx, scr.Name, scr.Namespace)
+			if err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to update keyType on spoke based on storagecluster %q version in namespace %q in managed cluster: %v", scr.Name, scr.Namespace, err)
+			}
+			if mirrorPeer.Annotations[spokeKeyTypeAnn] != spokeAnnVal {
+				mirrorPeer.Annotations[spokeKeyTypeAnn] = spokeAnnVal
```

Comment:
How about using a function like: https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/util/util.go#L82-L93 to add/update the annotation on localObject

We can do this in other places as well

Reply:
Not much of a help in this case as we are accessing the annotation before updating it.